### PR TITLE
Update rails6.1

### PR DESCRIPTION
**Pull request by bbarberBPL**
- Updated rails to recent 6.1 release 
- Updated all project dependencies
- Fixed error formatting errors in `application_controller.rb` due to new update
- Fixed ```DEPRECATION WARNING: Using `return`, `break` or `throw` to exit a transaction block is
deprecated without replacement. ``` in `find_or_create_nomenclature` method in `factory_service.rb`
- Changed `digital_origin` to be a postgres `ENUM TYPE`

### CODE REVIEW

File: app/models/curator/metastreams/descriptive.rb [hunk 1]
```diff
@@ -7,7 +7,13 @@ class Metastreams::Descriptive < ApplicationRecord
     include AttrJson::Record::Dirty
     include AttrJson::NestedAttributes
 
-    enum digital_origin: ['born digital', 'reformatted digital', 'digitized microfilm', 'digitized other analog'].freeze
+    enum digital_origin: {
```
Comment (ebenenglish) [inline app/models/curator/metastreams/descriptive.rb:10]: If the keys and values are the same, could this not be rewritten as:
```ruby
enum digital_origin: %w(born_digital reformatted_digital digitized_microfilm digitized_other_analog).freeze
```
Or can we keep the space in the values, but not the keys? As in:
```ruby
born_digital: 'born digital'
```

File: spec/factories/curator/metastreams/descriptives.rb [hunk 2]
```diff
@@ -14,7 +14,6 @@
     related { create(:curator_descriptives_related) }
     title { create(:curator_descriptives_title_set) }
     subject_other { create(:curator_descriptives_subject) }
```
Comment (ebenenglish) [inline spec/factories/curator/metastreams/descriptives.rb:16]: Should this be replaced with `digital_origin { 'reformatted_digital' }` ? Or is there an implied default so it's no longer needed?

File: spec/models/curator/metastreams/descriptive_spec.rb [hunk 3]
```diff
@@ -131,8 +131,8 @@
 
   describe 'Enum attributes' do
     it { is_expected.to define_enum_for(:digital_origin).
-                     with_values(['born digital', 'reformatted digital', 'digitized microfilm', 'digitized other analog']).
-                     backed_by_column_of_type(:integer) }
+                     with_values(born_digital: 'born_digital', reformatted_digital: 'reformatted_digital', digitized_microfilm: 'digitized_microfilm', digitized_other_analog: 'digitized_other_analog').
```
Comment (ebenenglish) [inline spec/models/curator/metastreams/descriptive_spec.rb:134]: See comment from app/models/curator/metastreams/descriptive.rb.

**Review by ebenenglish — COMMENTED** [review]
Looks solid, just a few questions about the new `digital_origin` enum stuff related to my own lack of knowledge about how these columns work in latest ActiveRecord, etc.

File: app/models/curator/metastreams/descriptive.rb (hunk 1 above)
Comment (bbarberBPL) [reply app/models/curator/metastreams/descriptive.rb:10]: We should keep them underscored. I can try to change it to an array

File: spec/factories/curator/metastreams/descriptives.rb (hunk 2 above)
Comment (bbarberBPL) [reply spec/factories/curator/metastreams/descriptives.rb:16]: It has that set as a default

**Pull request merged**